application crashes if in collections module (tag) any leaf element of the tree is selected

**Introduction**
see title

**Did you buy darktable from an application store ?**
no, self build from git master

**To Reproduce**
- Go to lighttable
- Goto collections module
- Select metadata -> tag (tree of tags correctly shown, selection shown)
- Select a **branch** node (branch correctly expanded/collapsed, selection shown)
- Select an arbitrary **leaf** node (as shown in screenshot) -> application crashes

**Expected behavior**
Selection is shown in lighttable, no crash

**Screenshots**
![collections-leaf-crash](https://user-images.githubusercontent.com/104291381/164958496-d707e4d4-bc94-4aeb-9bd5-3ab7a9b25809.png)

**Which commit introduced the error**
one of the large commits of the work in progress on selections

**Platform**
* darktable version : 3.9.0+1270~g1c28af863
* OS                : linux 5.4.0-109-generic
* Linux - Distro    : Ubuntu 20.04
* Memory            : 64 GB
* Graphics card     : Nvidia GeForce RTX 2060
* Graphics driver   : 470.57.01-0ubuntu0.20.04.3
* OpenCL installed  : yes
* OpenCL activated  : yes
* Xorg              : 
* Desktop           : LXDE
* GTK+              : 
* gcc               : 
* cflags            : 
* CMAKE_BUILD_TYPE  : 

**Additional context**
Backtrace and -d all debug output can be provided if necessary. It was not possible to upload the files, neither as txt files, nor as tar.gz or zip archive. 


 - Can you reproduce with another darktable version(s)? **did not try**
 - Can you reproduce with a RAW or Jpeg or both? **both**

I can reproduce too on master (with 3.9.0+1271).

@AlicVB: seems to be related to your collect filters huge PR. I tried a bisect and come inside. I didn't succeed to find commit. I'm stuck here:

```
Bissection : 61 révisions à tester après ceci (à peu près 6 étapes)
[d997f24b7ab9567c2ca58af1393e65957f3807c4] filtering : fix un-pin click + range graph in topbar
```
When I compile commit cited here, I have that error:
```

[ 40%] Building C object bin/CMakeFiles/lib_darktable.dir/common/dbus.c.o
[ 40%] Building C object bin/CMakeFiles/lib_darktable.dir/common/dlopencl.c.o
/home/nilvus/Projets/darktable/src/common/datetime.c: In function ‘dt_datetime_entry_to_exif’:
/home/nilvus/Projets/darktable/src/common/datetime.c:293:5: error: implicit declaration of function ‘dt_datetime_unix_lt_to_exif’; did you mean ‘dt_datetime_unix_to_exif’? [-Werror=implicit-function-declaration]
  293 |     dt_datetime_unix_lt_to_exif(exif, exif_len, &now);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |     dt_datetime_unix_to_exif
compilation terminated due to -Wfatal-errors.
cc1: all warnings being treated as errors
```
Same error with commit 2fb7d5b5679fb5e1b97af649e047b8a17443d3ad

Hope that will help you

--- a/src/control/signal.c
+++ b/src/control/signal.c
@@ -129,7 +129,7 @@ static dt_signal_description _signal_description[DT_SIGNAL_COUNT] = {
     G_CALLBACK(_image_info_changed_destroy_callback), FALSE }, // DT_SIGNAL_IMAGE_INFO_CHANGED
   { "dt-style-changed", NULL, NULL, G_TYPE_NONE, g_cclosure_marshal_VOID__VOID, 0, NULL, NULL,
     FALSE }, // DT_SIGNAL_STYLE_CHANGED
-  { "dt-images-order-change", NULL, NULL, G_TYPE_NONE, g_cclosure_marshal_VOID__UINT, 1, pointer_arg, NULL,
+  { "dt-images-order-change", NULL, NULL, G_TYPE_NONE, g_cclosure_marshal_generic, 1, pointer_arg, NULL,
     FALSE }, // DT_SIGNAL_IMAGES_ORDER_CHANGE
   { "dt-filmrolls-changed", NULL, NULL, G_TYPE_NONE, g_cclosure_marshal_VOID__VOID, 0, NULL, NULL,
     FALSE }, // DT_SIGNAL_FILMROLLS_CHANGED

